### PR TITLE
Global Bindings fixes and updates

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description

- Exposed new Global bindings for the Tableblock: `Rows`, `Extra Info`, `Rows Length`, `Schema`, `Page Number`
- `PageNumber` binding fix - General fix for the binding context data in the DataProvider component. An existing issue.
- Added a `scope` prop to the `Repeater` to allow block components to surface their row data globally.
  - In general, repeaters within blocks will only return 1 row so adding it globally makes sense.
  - The default repeater behaviour remains the same. Row data/context is scoped locally to each repeater row.
- Formblock/Multistep Formblock - allow access to internal row data
  - Used the new repeater flag to switch the internal repeater context to global

## Addresses
- https://github.com/Budibase/budibase/pull/11830

### CODE REVIEW

File: packages/client/src/components/app/Repeater.svelte [hunk 1]
```diff
@@ -22,7 +23,10 @@
     <Placeholder />
   {:else if rows.length > 0}
     {#each rows as row, index}
-      <Provider data={{ ...row, index }} scope={ContextScopes.Local}>
+      <Provider
+        data={{ ...row, index }}
+        scope={nested ? ContextScopes.Global : ContextScopes.Local}
```
Comment (shogunpurple) [inline packages/client/src/components/app/Repeater.svelte:28]: Is there any reason this can't always just be global?

Comment (shogunpurple) [reply packages/client/src/components/app/Repeater.svelte:28]: Ah that makes sense - we would need a new way to key the repeater rows in the global context. Is `global` a better name for this property then?

Comment (deanhannigan) [reply packages/client/src/components/app/Repeater.svelte]: Yep agreed. It does step on the current nested flag used elsewhere in the block components. I've just replaced it with a configurable scope prop instead.

**Review by shogunpurple — APPROVED** [review]
LGTM

**Pull request merged**